### ISSUE TITLE
httpserver doesn't serve external clients

### ISSUE THREAD

**Issue by WoodyAtHome** [ponylang/http#19]
Hi,

I'm not sure if I have an issue for ponylang/http or my network or raspbian configuration.

I compiled the httpserver example an run it on my raspbian stretch:

`pi@homeserver:~/src/.deps/ponylang/http/examples/httpserver $ sudo ./httpserver 8080
connected: ::1
`

It runs fine, a test with wget shows:
 ```
pi@homeserver:~/src$ wget localhost:8080/Hello
--2019-03-21 11:40:22--  http://localhost:8080/Hello
Auflösen des Hostnamens »localhost (localhost)« … ::1, 127.0.0.1
Verbindungsaufbau zu localhost (localhost)|::1|:8080 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 200 OK
Länge: 20
Wird in »»Hello«« gespeichert.

Hello               100%[===================>]      20  --.-KB/s    in 0s      

2019-03-21 11:40:22 (645 KB/s) - »»Hello«« gespeichert [20/20]
```

But when I try to access from another machine (in the same network), it's not possible (the Raspberry is named 'homeserver.local'):
```
woody@lappi:~$ wget homeserver.local:8080/Hello
--2019-03-21 11:55:43--  http://homeserver.local:8080/Hello
Auflösen des Hostnamens homeserver.local (homeserver.local) … 192.168.0.236
Verbindungsaufbau zu homeserver.local (homeserver.local)|192.168.0.236|:8080 … fehlgeschlagen: Verbindungsaufbau abgelehnt.
```

I also installed lighttp on the Raspberry, no problem to access it from another machine:
```
woody@lappi:~$ wget homeserver.local
--2019-03-21 11:57:35--  http://homeserver.local/
Auflösen des Hostnamens homeserver.local (homeserver.local) … 192.168.0.236
Verbindungsaufbau zu homeserver.local (homeserver.local)|192.168.0.236|:80 … verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 OK
Länge: 3378 (3,3K) [text/html]
Wird in »index.html« gespeichert.

index.html          100%[===================>]   3,30K  --.-KB/s    in 0s      

2019-03-21 11:57:35 (16,4 MB/s) - »index.html« gespeichert [3378/3378]
```

I can not see any reason why the Raspberry refuses the connection to httpserver.
```
pi@homeserver:~/src $ sudo iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination         

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination         

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         
```

```
pi@homeserver:~/src $ sudo netstat -tlnpu
Aktive Internetverbindungen (Nur Server)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      1368/lighttpd       
tcp        0      0 0.0.0.0:8083            0.0.0.0:*               LISTEN      463/perl            
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      457/sshd            
tcp6       0      0 ::1:8080                :::*                    LISTEN      4257/./httpserver   
tcp6       0      0 :::80                   :::*                    LISTEN      1368/lighttpd       
tcp6       0      0 :::22                   :::*                    LISTEN      457/sshd            
udp        0      0 0.0.0.0:57862           0.0.0.0:*                           381/avahi-daemon: r 
udp        0      0 0.0.0.0:68              0.0.0.0:*                           353/dhcpcd          
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           381/avahi-daemon: r 
udp6       0      0 :::60888                :::*                                381/avahi-daemon: r 
udp6       0      0 :::546                  :::*                                353/dhcpcd          
udp6       0      0 :::5353                 :::*                                381/avahi-daemon: r 
```

Just an idea..: The httpserver is only listed as tcp6 listening. Maybe that could be the problem?

Thanks for any help, I'm not so familiar with the network stuff.

**Comment by mfelsche** [comment]
Hi @WoodyAtHome thanks for the issue. As your `netstat` output shows, the tcp example server only listens on `::1`. The reason for this is that the example server has `localhost` hard-coded as host to listen on. This can be changed easily.
When the pony runtime resolved `localhost`, it takes the first address it resolved to and starts listening on that. As in your wget call above, `localhost` resolves to: `::1, 127.0.0.1` in that order.
Your lighttpd listens to "0.0.0.0", so you can reach it from outside.
So, if you change the example before you start it, to have the empty string as the host, it should resolve to the wildcard IP address ("0.0.0.0" for IPv4).

**Comment by WoodyAtHome** [comment]
Hi @mfelsche , you are right, with `host = ""` everything is all right and works!

Thanks a lot!
